src/main/kotlin/no/nav/familie/oppdrag/simulering/SimulerBeregningRequestMapper.kt

```diff
@@ -0,0 +1,93 @@
+
+    fun tilSimulerBeregningRequest(utbetalingsoppdrag: Utbetalingsoppdrag): no.nav.system.os.tjenester.simulerfpservice.simulerfpservicegrensesnitt.SimulerBeregningRequest =
+            fpServiceGrensesnittFactory.createSimulerBeregningRequest()
+                    .apply { request = tilSimulerBeregningTypesRequest(utbetalingsoppdrag) }
+
+    private fun tilSimulerBeregningTypesRequest(utbetalingsoppdrag: Utbetalingsoppdrag): no.nav.system.os.tjenester.simulerfpservice.simulerfpserviceservicetypes.SimulerBeregningRequest =
+            fpServiceTypesFactory.createSimulerBeregningRequest().apply { oppdrag = tilOppdrag(utbetalingsoppdrag) }
+
+    private fun tilOppdrag(utbetalingsoppdrag: Utbetalingsoppdrag): Oppdrag {
+        oppdragSkjemaFactory.createAvstemmingsnokkel().apply {
+            avstemmingsNokkel = utbetalingsoppdrag.avstemmingTidspunkt.format(tidspunktFormatter)
+            kodeKomponent = utbetalingsoppdrag.fagSystem
+            tidspktReg = utbetalingsoppdrag.avstemmingTidspunkt.format(tidspunktFormatter)
+        }
+
+        val oppdragsEnhet = oppdragSkjemaFactory.createEnhet().apply {
+            enhet = OppdragSkjemaConstants.ENHET
+            typeEnhet = OppdragSkjemaConstants.ENHET_TYPE
+            datoEnhetFom = OppdragSkjemaConstants.ENHET_DATO_FOM.toString()
+        }
+
+        return fpServiceTypesFactory.createOppdrag().apply {
+            kodeEndring = EndringsKode.fromKode(utbetalingsoppdrag.kodeEndring.name).kode
+            kodeFagomraade = utbetalingsoppdrag.fagSystem
+            fagsystemId = utbetalingsoppdrag.saksnummer
+            utbetFrekvens = UtbetalingsfrekvensKode.MÅNEDLIG.kode
+            oppdragGjelderId = utbetalingsoppdrag.aktoer
+            datoOppdragGjelderFom = OppdragSkjemaConstants.OPPDRAG_GJELDER_DATO_FOM.toString()
+            saksbehId = utbetalingsoppdrag.saksbehandlerId
+            enhet.add(oppdragsEnhet)
+            avstemmingsnokkel.addAll(avstemmingsnokkel)
+            //ompostering
+            utbetalingsoppdrag.utbetalingsperiode.map { periode ->
+                oppdragslinje.add(tilOppdragsLinje(utbetalingsperiode = periode, utbetalingsoppdrag = utbetalingsoppdrag))
+            }
+        }
+    }
+
+    private fun tilOppdragsLinje(utbetalingsperiode: Utbetalingsperiode, utbetalingsoppdrag: Utbetalingsoppdrag): Oppdragslinje {
+
+        val attest = oppdragSkjemaFactory.createAttestant().apply {
+            attestantId = utbetalingsoppdrag.saksbehandlerId
+        }
+
+        return fpServiceTypesFactory.createOppdragslinje().apply {
+            kodeEndringLinje =
+                    if (utbetalingsperiode.erEndringPåEksisterendePeriode) EndringsKode.ENDRING.kode else EndringsKode.NY.kode
+            utbetalingsperiode.opphør?.let {
+                kodeStatusLinje = KodeStatusLinje.OPPH
+                datoStatusFom = it.opphørDatoFom.toXMLDate().toString()
```

Comment:
Skal opphørsdatoen konverteres til XMLDate? Det er ikke gjort for datoene på 110-nivå. Regner med at det er samme format som gjelder i alle datofelt?